### ISSUE TITLE
How the patch size and input size are set with the Davis mask propagation experiment ?

### ISSUE THREAD

**Issue by jrabary** [facebookresearch/dino#56]
Hi, 
Thanks for sharing your implementation of dino.

We are trying to figure out how do you set the size of the patches when evaluating on Davis. 
You said in the paper that the input resolution is 480 for Davis and images are not square.
Or, the VIT model seems to be pre-trained with an input resolution fo 224 and patch size of 16. 
So there is a kind of sequence length mismatch between the pre-training and the test.

Can you give a more detail on this ?

**Comment by mathildecaron31** [comment]
Hi @jrabary 
Thanks for your interest

You can set the patch size with the following argument: https://github.com/facebookresearch/dino/blob/4b96393c4c877d127cff9f077468e4a1cc2b5e2d/eval_video_segmentation.py#L256

There is indeed a sequence length mismatch between pre-training and testing since image resolutions are different. However the patch size should be the same (16x16 or 8x8). Hope that helps.

**Comment by jrabary** [comment]
Hi @mathildecaron31 

Thank you for your reply.

So, if I understand it, to fill the gap between the pre-training and test resolution you interpolate the positional encoding ?

**Comment by GerardMaggiolino** [comment]
@jrabary Not an author or contributor, so I'd definitely verify what I'm sharing, but the paper states they use the DeiT implementation, which in turn defaults to the ViT architecture. In the ViT paper, it's stated that 2D interpolation of the position embeddings are used according to the patch / embedding locations in the original image. 

There's also [this code directly from the repo](https://github.com/facebookresearch/dino/blob/4b96393c4c877d127cff9f077468e4a1cc2b5e2d/vision_transformer.py#L174) which appears to implement that.

**Comment by jrabary** [comment]
Thank you @GerardMaggiolino. Yes, this is what I understand after re-reading the papers and checking the code. But I was unsure :-)

**Comment by mathildecaron31** [comment]
Hi @jrabary and @GerardMaggiolino 

Yes, we are doing bicubic interpolation of positional encodings: https://github.com/facebookresearch/dino/blob/4b96393c4c877d127cff9f077468e4a1cc2b5e2d/vision_transformer.py#L174
